### PR TITLE
⚡️ feat: add performance INFO in logs

### PR DESCRIPTION
**Pull request by peb-peb**
## Description

- added performance INFO in logs for the project using the `time` library's `perf_counter()` and `process_time()`.
  - `perf_counter()` - measures the time since the program started
  - `process_time()` - measures the time spent by the current process executing code

- **Results: (sample space = 10)**
  - Average Elapsed time since the program started: **146.262744**
  - Average Elapsed time spent by the current process executing code: **4.424605**

**NOTE:** [related docs for above topic](https://docs.python.org/3/library/time.html#time.perf_counter)

## Changes Made
- `scanner.py` - added performance INFO in logs

## Checklist
- [x] I have read and followed the contributing guidelines.
- [x] I have tested my changes thoroughly and they work as expected.
- [x] I have added necessary tests for the changes made.
- [ ] I have updated the documentation to reflect the changes made.
- [x] My code follows the project's coding style and standards.
- [x] I have added appropriate commit messages and comments for my changes.

## Additional Notes
The goal is to improve the "Average Elapsed time since the program started" time significantly.

### CODE REVIEW

**Review by mshudrak — COMMENTED** [review]
would it make more sense to have something like debug/profiling flag to turn this on only when required (e.g. in tests)?

**Comment by peb-peb** [comment]
I'll implement it.

**Comment by peb-peb** [comment]
image

Currently, added the `--time` flag for "performance testing".

**Comment by peb-peb** [comment]
image

added tests for the `--time` params. (using pytest fixtures)
- [SKIP] If the parameter is not passed in the test.
- [PASS] It checks whether the value of the parameter is `True` as intended.

Also, should I add these tests for all parameters?

**Comment by mshudrak** [comment]
Yes, it would be great to have them for all parameters in the future.

File: src/gcp_scanner/test_params.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+def test_name(perf_time):
+  assert perf_time == True
```
Comment (mshudrak) [inline src/gcp_scanner/test_params.py:2]: Nit: add new line.

File: src/gcp_scanner/conftest.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+def pytest_addoption(parser):
+  parser.addoption("--time", action="store_true")
+
+@pytest.fixture()
```
Comment (mshudrak) [inline src/gcp_scanner/conftest.py:6]: I'd probably add a blurb that we should expand conf test to other tests too.

Comment (mshudrak) [reply src/gcp_scanner/conftest.py:6]: We will likely have only one set of them for now but we can expand it to cover various other configs e.g. fetching creds from metadata vs service account key and etc.

**Review by mshudrak — APPROVED** [review]
Overall looks good, I have some minor comments on what to add. I would advice to expand tests for other orgs in a separate PR.

**Comment by mshudrak** [comment]
oops, forgot to take a look but pylint is complaining about several places. Please fix.

**Comment by peb-peb** [comment]
fixed the `pylint` errors and the changes requested.

**Comment by peb-peb** [comment]
> =========================== short test summary info ============================
> ERROR src/gcp_scanner/test_params.py::test_name - AttributeError: 'Namespace' object has no attribute 'time'

It is failing because it is unable to find the `time` attribute/parameter (which is being added in this PR).
It would `PASS` - if the time parameter is passed or `SKIP` - if the time parameter is absent in tests, once this PR is merged.

**Pull request merged**